src/components/Modules/ModuleText.js

```diff
@@ -33,25 +27,27 @@ export default function ModuleText() {
   const user = useSelector(state => state.userReducer);
   const dispatch = useDispatch();
   const { push } = useHistory();
+  const { courseId, moduleId } = useParams();
 
   useEffect(() => {
     if (status === 'delete/success') {
       // TODO needs support for new modules path
       // import {pathUtils} from "../../routes";
       // use pathUtils.makeViewAllModulesPath function
-      push(pathUtils.makeViewAllModulesPath());
-      push('/modules');
+      // grab course id from use params hook
+      dispatch(moduleActions.deleteModuleThunk(module.moduleid));
+      push(pathUtils.makeViewAllModulesPath(courseId));
     }
   }, [push, status]);
 
   const handleMenuClick = e => {
     if (e.key === 'edit') {
-      dispatch(moduleActions.setEditModule(currentModule));
+      dispatch(moduleActions.getModuleByModuleIdThunk(module.moduleid));
       // TODO needs support for new edit-module path — import {pathUtils} from "../../routes";
       // use pathUtils.makeEditModulePath(moduleId)
-      push('/edit-module');
+      push(pathUtils.makeEditModulePath(moduleId));
```

Comment:
No, we did not discuss this, but we should... @simonesquad @sebass482 @rhea-manuel and Nam and Marina(sorry, can't "at" because I can't find your GitHub handle) It's not a huge deal, but I vote for the MenuItemLink component we built out together. See `src/components/Program/ProgramCard.js` lines 29 through 40 and `src/components/_common/MenuItemLink.js` for reference. Or reach out and we can go through it as a team.